File: Easy/Pascal Triangle/pascal-triangle.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// #define ll long long
class Solution{
public:
    vector<long long> nthRowOfPascalTriangle(int n) {
        // code here
        // long long ans=1;
        // vector<long long> l;
        // n=n-1;
        // int m=(pow(10,9))+7;
        // //l.push_back(ans);
        // for(int i=0;i<=n;i++){
        //     l.push_back(ans);
        //     ans=((ans*(n-i))/(i+1))%m;
        //     //ans=ans/(i+1);
        // }
        // return l;
        long long mod=1e9+7;
        vector<vector<long long>>ans(n);
        for(int i=0;i<n;i++)
        {
            ans[i].resize(i+1);
            ans[i][0]=ans[i][i]=1;
            
            for(int j=1;j<i;j++)
            {
                ans[i][j]=(ans[i-1][j-1]+ans[i-1][j])%mod;
            }
        }
        return ans[n-1];
    

    }
};


//{ Driver Code Starts.


void printAns(vector<long long> &ans) {
    for (auto &x : ans) {
        cout << x << " ";
    }
    cout << "\n";
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution ob;
        auto ans = ob.nthRowOfPascalTriangle(n);
        printAns(ans);
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Medium/Pascal Triangle/pascal-triangle.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());

            ArrayList<Integer> ans = new Solution().nthRowOfPascalTriangle(n);
            printAns(ans);

            System.out.println("~");
        }
    }

    public static void printAns(ArrayList<Integer> ans) {
        for (Integer x : ans) {
            System.out.print(x + " ");
        }
        System.out.println();
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {

    ArrayList<Integer> nthRowOfPascalTriangle(int n) {
        // code here
        ArrayList<Integer> a1=new ArrayList<>();
        a1.add(1);
        for(int i=2;i<=n;i++){
           // System.out.println(a1);
            ArrayList<Integer> a2=new ArrayList<>();
            for(int j=1;j<=i;j++){
                if(j==1 || j==i){
                    a2.add(1);
                }
                else{
                    a2.add(a1.get(j-2)+a1.get(j-1));
                }
            }
            a1=a2;
        }
        return a1;
    }
    
}